client/python/rappor.py

```diff
@@ -119,39 +119,37 @@ def from_csv(f):
     return p
 
 
-class _SimpleRandom(object):
+class _SecureRandom(object):
   """Returns an integer where each bit has probability p of being 1."""
 
-  def __init__(self, prob_one, num_bits, _rand=None):
+  def __init__(self, prob_one, num_bits):
     self.prob_one = prob_one
     self.num_bits = num_bits
-    self._rand = _rand or random.Random()
 
   def __call__(self):
     p = self.prob_one
-    rand_fn = self._rand.random  # cache it for speed
+    bytes = os.urandom(self.num_bits)
 
     r = 0
-    for i in xrange(self.num_bits):
-      bit = rand_fn() < p
+    for i, b in enumerate(xrange(self.num_bits)):
+      bit = b < p * 255.0
```

Comment:
1. Done.
2. I did too. However the interface provided by the secure random number generator is only bytes. It is not worth having code to do it in a nicer way if nobody can use it in production. The real question is how many bits of precision do we need. I can easily change this code to support 1/2^16th or 1/2^32 precision. (Which is effectively what floating point would be doing anyway.) (at the cost of a 2x or 4x slowdown respectively) While the cost is low enough to be acceptable; I'm not sure it adds a lot of value. Do we really need to specify values of P and Q to more than one part in 256?